dashboard/api_edx_cache.py

```diff
@@ -0,0 +1,264 @@
+"""
+APIs that deal with the edx cached data
+"""
+import datetime
+import logging
+
+from django.db import transaction
+import pytz
+
+from courses.models import CourseRun
+from dashboard import models
+from profiles.api import get_social_username
+from search import tasks
+
+log = logging.getLogger(__name__)
+
+
+class CachedEdxUserData:
+    """
+    Class to handle the retrieval and update of the users' cached edX information
+    """
+
+    ENROLLMENT = 'enrollment'
+    CERTIFICATE = 'certificate'
+    CURRENT_GRADE = 'current_grade'
+
+    # the sorting of the supported caches matters for refresh
+    SUPPORTED_CACHES = (ENROLLMENT, CERTIFICATE, CURRENT_GRADE,)
+
+    CACHED_EDX_MODELS = {
+        ENROLLMENT: models.CachedEnrollment,
+        CERTIFICATE: models.CachedCertificate,
+        CURRENT_GRADE: models.CachedCurrentGrade,
+    }
+
+    CACHE_EXPIRATION_DELTAS = {
+        ENROLLMENT:  datetime.timedelta(minutes=5),
+        CERTIFICATE: datetime.timedelta(hours=6),
+        CURRENT_GRADE: datetime.timedelta(hours=1),
+    }
+
+    @classmethod
+    def get_cached_edx_data(cls, user, cache_type):
+        """
+        Helper function to return cached data.
+
+        Args:
+            user (django.contrib.auth.models.User): A user
+            cache_type (str): a string representing one of the cached data types
+        Returns:
+            Enrollments or Certificates or CurrentGrades
+        """
+        if cache_type not in cls.SUPPORTED_CACHES:
+            raise ValueError("{} is an unsupported cache type".format(cache_type))
+        return cls.CACHED_EDX_MODELS[cache_type].get_edx_data(user)
+
+    @classmethod
+    def update_cache_last_access(cls, user, cache_type, timestamp=None):
+        """
+        Updates the UserCacheRefreshTime model timestamp for the provided cache type
+
+        Args:
+            user (django.contrib.auth.models.User): A user
+            cache_type (str): a string representing one of the cached data types
+            timestamp (datetime): a timestamp
```

Comment:
Just being pedantic but I think you want `datetime.datetime` here for the arg type